.travis.yml

```diff
@@ -5,50 +5,76 @@ cache:
-
-# 1. Let Travis automatically create a matrix of all combinations of PHP versions
-# (the ones listed under `php`), and environments (listed under `env`) and test
-# against all those combinations.
-
-# Test main supported versions of PHP against latest WP.
-php:
-  - 7.0
-  - 7.1
-  - 7.2
-  - 7.3
-
 services:
   - mysql
 
 # Different stages in order to only deploy when everything succeeds
 stages:
   - test
+  - name: compatibility
+    if: branch = master
+  - name: compatibility-edge
+    if: branch = master
   - e2e-testing
 
 env:
-  - WP_VERSION=latest WC_VERSION=4.3.0
+  global:
+    - WC_NEWEST_SUPPORTED_VERSION=4.6.0
+    - WC_MOST_USED_VERSION=4.5.0
+    - WC_OLDEST_SUPPORTED_VERSION=4.0.0
+    - WP_OLDEST_SUPPORTED_VERSION=5.3
 
 # 2. Also test against these combinations, as listed out explicitly one by one
 
 # Additional tests against stable PHP (min recommended version is 5.6) and past supported versions of WP
 # and code coverage report.
 matrix:
   fast_finish: true
+  allow_failures:
+    - php: 7.4
+      env: WP_VERSION=nightly WC_VERSION=latest
+      stage: compatibility-edge
   include:
-  - name: "Coding standard check"
-    php: 7.2
-    env: WP_VERSION=latest WC_VERSION=4.3.0 RUN_PHPCS=1
+# PHP versions compatibility (latest WordPress and most used WooCommerce versions)
+  - php: 7.0
+    env: WP_VERSION=latest WC_VERSION=$WC_MOST_USED_VERSION RUN_PHPCS=1
+    stage: test
+  - php: 7.1
+    env: WP_VERSION=latest WC_VERSION=$WC_MOST_USED_VERSION
+    stage: test
+  - php: 7.2
+    env: WP_VERSION=latest WC_VERSION=$WC_MOST_USED_VERSION
     stage: test
   - php: 7.3
-    env: WP_VERSION=latest WC_VERSION=4.3.0
+    env: WP_VERSION=latest WC_VERSION=$WC_MOST_USED_VERSION
     stage: test
-  - name: "PHP 7.3 unit tests using WordPress nightly"
-    php: 7.3
-    env: WP_VERSION=nightly WC_VERSION=4.3.0
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=$WC_MOST_USED_VERSION
     stage: test
+# bottom compatibility edge (oldest supported PHP/WordPress/WooCommerce versions)
+  - php: 7.0
+    env: WP_VERSION=$WP_OLDEST_SUPPORTED_VERSION WC_VERSION=$WC_OLDEST_SUPPORTED_VERSION
+    stage: compatibility-edge
+# WooCommerce versions compatibility (latest supported PHP/WordPress versions on target WooCommerce versions)
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=$WC_OLDEST_SUPPORTED_VERSION
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=4.4.0
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=$WC_MOST_USED_VERSION
+    stage: compatibility
+  - php: 7.4
+    env: WP_VERSION=latest WC_VERSION=$WC_NEWEST_SUPPORTED_VERSION
+    stage: compatibility
+# top compatibility edge (nightly versions of WordPress on latest supported WooCommerce/PHP version)
+  - php: 7.4
+    env: WP_VERSION=nightly WC_VERSION=latest
+    stage: compatibility-edge
+# E2E testing
   - name: "E2E tests"
+    env: WP_VERSION=latest WC_VERSION=$WC_MOST_USED_VERSION
```

Comment:
IIRC `WC_VERSION` is ignored by setup script here, but it should be addresses in another PR.

Reply:
I'll take a look =)